ietf/doc/models.py

```diff
@@ -642,9 +642,23 @@ def pdfized(self):
         return pdf
 
     def references(self):
+        from ietf.doc.utils import rebuild_reference_relations
+        files = {}
+        for type in ["xml", "txt"]:
+            name = self.get_file_name().replace(".txt", f".{type}")
+            if os.path.isfile(name):
+                files[type] = name
+        rebuild_reference_relations(self, files)
```

Comment:
This adds a non-trivial penalty on _every access_ to the function. I was anticipating we would use a management command to update the database a single time (or at least on-demand) with a process other than the webserver.

I had also intended to update rebuild_reference_relations fully expand xincludes before extracting references from the XML (rather than the proposal to guess based on what the xinclude url contains).

Reply:
Argh. That was not supposed to be part of the commit - I just did this foe testing. Will remove.